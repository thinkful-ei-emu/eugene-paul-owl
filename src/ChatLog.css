.chat-log{
  vertical-align:baseline;
  overflow:scroll;
}

.chat-form{
  vertical-align:baseline;
  padding:10px;
}

.chat-form-input{
  width:100%;
}

.chat-flex-wrapper{
  display:flex;
  flex-direction:column;
  justify-content:flex-end;
  height:100%;
  border: 1px solid rgb(223, 218, 218);
}