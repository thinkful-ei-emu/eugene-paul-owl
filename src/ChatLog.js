import React from 'react';
import ChatMessage from './ChatMessage';

function ChatLog(props){
  let chatList=props.chatEvents.map((event)=>{
    let id=event.participantId;
    let part=props.participants.find((part)=>id===part.id);  

    return <ChatMessage id={id} name={part.name} avatar={part.avatar} type={event.type} message={event.message} time={event.timestamp} />
  }
  )
  console.log(chatList);

  return(
    <div> 
      <div>
        {chatList}
      </div>
      <form className="chat-form">
        <textarea className="chat-form-input" placeholder="Chat" style={{height: '45px'}}>
        </textarea>
        <input type="submit" className="chat-form-submit"/>

      </form>
    </div>
  );
}

export default ChatLog;