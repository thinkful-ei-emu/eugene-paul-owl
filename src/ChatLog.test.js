import React from 'react';
import ReactDOM from 'react-dom';
import ChatLog from './ChatLog';
import renderer from 'react-test-renderer';
import {chatEvents, participants} from './Store';

console.log(chatEvents);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatLog chatEvents={chatEvents} participants={participants}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('renders UI with default props', () => {
  const tree = renderer.create(<ChatLog chatEvents={chatEvents} participants={participants}/>);
  expect(tree).toMatchSnapshot();
});