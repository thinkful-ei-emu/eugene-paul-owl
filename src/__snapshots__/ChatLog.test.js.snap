// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI with default props 1`] = `
<div>
  <div>
    <div
      className="message"
    >
      <img
        alt="avatar"
        className="user-avatar message-avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <div
        className="message-body"
      >
        <h6
          className="message-body-header"
        >
          Axe Kubicka
        </h6>
        <span
          className="message-body-time"
        >
           
          1548852484247
           
        </span>
      </div>
      <div
        className="message-body-content"
      >
        <p>
          Hello world
        </p>
      </div>
    </div>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
  <form
    className="chat-form"
  >
    <textarea
      className="chat-form-input"
      placeholder="Chat"
      style={
        Object {
          "height": "45px",
        }
      }
    />
    <input
      className="chat-form-submit"
      type="submit"
    />
  </form>
</div>
`;
